R/map_vis.R

```diff
@@ -0,0 +1,200 @@
+#' plot_watershed
+#'
+#' @param data_dir root directory for the spatial data ("/pic/projects/im3/teleconnections/data/")
+#' @param watersheds_file_path path of watersheds shapefile within data_dir
+#' @param landcover_file_path path of NLCD landcover data file
+#' @param powerplants_file_path path of power plants data file
+#' @param city character. City to be analysed in the format "City | STATE" (e.g., "Phoenix | AZ")
+#' @details plots the watersheds and relevant features for a chosen city
+#' @import tmap
+#' @importFrom tmaptools aggregate_map
+#' @importFrom dplyr filter mutate if_else right_join
+#' @importFrom tibble rowid_to_column
+#' @importFrom raster levels extent
+#' @export
+#'
+plot_watershed <- function(data_dir,
+                           watersheds_file_path = "water/CWM_v2_2/World_Watershed8.shp",
+                           landcover_file_path = "land/NLCD_2016_Land_Cover_L48_20190424/NLCD_2016_Land_Cover_L48_20190424.img",
+                           crop_file_path = "land/2016_30m_cdls/2016_30m_cdls.img",
+                           powerplants_file_path = "water/UCS-EW3-Energy-Water-Database.xlsx",
+                           conus_file_path = "misc/states_21basic/states.shp",
+                           city){
+
+  all_cities <- get_cities()[["city_state"]]
+
+  # throw error if any chosen city lies outside available cities
+  if(!city %in% all_cities) {
+    stop(paste0(paste(city), " is not part of 'teleconnect'!"))
+  }
+
+  # get city watershed_ids
+  get_cities() %>%
+    filter(city_state == !!city) ->
+    city_watershed_mapping
+
+  # read shapefiles for watersheds
+  import_shapefile(paste0(data_dir, watersheds_file_path),
+                   method = "rgdal") %>%
+    # subset for desired watersheds
+    subset(DVSN_ID %in% city_watershed_mapping$DVSN_ID) ->
+    watersheds_city
+
+  import_shapefile("../spatial data/water/CWM_v2_2/City_Centroid.shp",
```

Comment:
ah good spot. This isn't currently being used in the function actually, since the cities in this shape file don't match neatly with the city names in the other shape files. Possible future task for @KristianNelson !!